ctapipe/io/eventsource.py

```diff
@@ -208,6 +208,14 @@ def subarray(self) -> SubarrayDescription:
 
         """
 
+    @property
+    @abstractmethod
+    def obs_config(self) -> list[ObservationConfigurationContainer]:
```

Comment:
This type annotation is only supported from 3.9, we still support 3.8. Use `from typing import List; List[ObservationConfigurationContainer]`. Since this also should be immutable, I'd prefer `Tuple[ObservationConfiguration]`